File: leetcode/118 Pascal's Triangle/leetcode118.py
class Solution:
    # @param {integer} numRows
    # @return {integer[][]}
    def generate(self, numRows):
        if numRows==0:
            return []
        l=[1]
        r=[l]
        if(numRows==1):
            return r
        i=2
        while i<=numRows:
            temp=list(r[-1])
            temp.append(1)
            for k in range(i-2,0,-1):
                temp[k]=temp[k]+temp[k-1]
            r.append(temp)
            i=i+1
        return r

File: leetcode/118 Pascal's Triangle/leetcode118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ret;
        for(int i=0;i<numRows;i++){
            ret.push_back(vector<int>(i+1));
            ret[i][0]=1;
            for(int j=1;j<i;j++)
                ret[i][j]=ret[i-1][j-1]+ret[i-1][j];
            ret[i][i]=1;
        }
        return ret;
    }
};